### ISSUE TITLE
Manual upload of cover picture

### ISSUE THREAD

**Issue by imclint21** [go-shiori/shiori#994]
Hi,

For some bookmarks I would be able to upload a custom image when the automatic fetching does not works.

<img width="684" alt="image" src="https://github.com/user-attachments/assets/c0ea61b6-19b0-4713-b824-7c71635ded32">

**Comment by mayur1377** [comment]
hey @imclint21  , I can give this a try :)

**Comment by imclint21** [comment]
would be awesome!

**Comment by mayur1377** [comment]
the way the images are stored is quite inconsistent , make it difficult to work with :)
made a [workaround](https://github.com/go-shiori/shiori/compare/master...mayur1377:shiori:feature/image?expand=1) for have a "random" placeholder if it was unable to fetch images from the provided URL , 
lemme know if I can go ahead and raise a PR :)

**Comment by imclint21** [comment]
I would recommend to use https://place-hold.it/ or something neutral for default image, 

Also i'm not maintainer of this project :D

**Comment by mayur1377** [comment]
@fmartingr , lemme know what to do :D

**Comment by fmartingr** [comment]
Unsure what the ask is here. It is uploading thumbnails or it is having placeholders for bookmarks without thumbnails?

@imclint21 As a workaround, you can put an image into the `$SHIORI_DIR/thumb` folder named after the ID of the bookmark. That is, if you upload an image file named `44` in that folder it should show up as thumbnail. This may change recently if we normalize data in the database but should work as of today.

**Comment by imclint21** [comment]
You can edit all the metadata of any bookmark why not let the possibility to upload a custom thumb like I can edit the title?

I'm sure @mayur1377 can handle it quickly, just need to upload the new image in `$SHIORI_DIR/thumb` as you said.

**Comment by imclint21** [comment]
Any news on this @fmartingr @mayur1377? :D

**Comment by morley461** [comment]
That would be awesome. It's the only thing that's missing for me.

**Comment by imclint21** [comment]
I agree and it will give more  consistence, all can be edited excepting cover, while automatic cover downloader can don't do his job.

**Comment by imclint21** [comment]
Any news on this @fmartingr?

**Comment by fmartingr** [comment]
This is a low priority for me at the moment, more so considering that the workaround of putting the files manually in the data directory can be used.
